### PR TITLE
Only allow admins and supervisors of the same org to show or hide contacts

### LINKED ISSUE #4065: Limit case_assignment controller method permissions to the case assignment's organization
**What type(s) of user does this feature affect?**
 - volunteers
 - supervisors
 - admins

**Description**
A case assignment links a casa case to a volunteer user

Make sure all the methods in [`app/policies/case_assignment_policy.rb `](https://github.com/rubyforgood/casa/blob/main/app/policies/case_assignment_policy.rb ) contain some logic to prevent casa case assignment access from users belonging to a different organization. 

Fix / Update tests if needed

See [`app/policies/application_policy.rb`](https://github.com/rubyforgood/casa/blob/main/app/policies/application_policy.rb) for helper methods
See https://github.com/varvet/pundit

_QA Login Details:_  
[Link to QA site](https://casa-qa.herokuapp.com/)  

Login Emails: 
- volunteer1@example.com  view site as a volunteer
- supervisor1@example.com view site as a supervisor
- casa_admin1@example.com view site as an admin

password for all users: 12345678  

### Questions? Join Slack!

We highly recommend that you join us in slack https://rubyforgood.herokuapp.com/ #casa channel to ask questions quickly and hear about office hours (currently Tuesday 6-8pm Pacific), stakeholder news, and upcoming new issues.

### PR DESCRIPTION
**Pull request by mateusdeap**
### What github issue is this PR for, if any?
Resolves #4065.

### What changed, and why?
We tentatively altered the `show_or_hide_contacts?` method to only check if the user is an admin or supervisor and if they are in the same org as the casa case.

The reason is that the implementation added in #3637 seemed to only return true, since it or'ed `hide_contacts?` and `!hide_contacts?`

We might be wrong somehow, but decided to try and will revert if we break anything.

### How will this affect user permissions?
- Volunteer permissions:
  - Apparently, no changes
- Supervisor permissions:
  - Can only change contact exhibition if in the same org
- Admin permissions:
  - Can only change contact exhibition if in the same org

### How is this tested? (please write tests!) 💖💪
We added specs to `case_assignment_policy_spec.rb`

### Feelings gif (optional)
confused...

### CODE REVIEW

**Review by compwron — APPROVED** [review]
seems like an improvement 😅

File: app/policies/case_assignment_policy.rb [hunk 1]
```diff
@@ -29,9 +29,11 @@ def hide_contacts?
   end
 
   def show_or_hide_contacts?
-    hide_contacts? || !hide_contacts?
+    admin_or_supervisor? && same_org?
   end
 
+  private
```
Comment (JuanVqz) [inline app/policies/case_assignment_policy.rb:35]: :rocket: :1st_place_medal:

**Comment by JuanVqz** [comment]
> seems like an improvement sweat_smile

what do you mean?
As far as I understand [we were required to change a method per PR.](https://github.com/rubyforgood/casa/issues/4065#issuecomment-1284046055)

So, this PR is implementing the `same_org?` method in [def show_or_hide_contacts?](https://github.com/rubyforgood/casa/pull/4132/files#diff-1e00963bb42432cb0649cf85431cb8b0a5ec72528fa403571a247aa299464644R31) but I don't know why @mateusdeap changed the rest of the logic there.

**Comment by mateusdeap** [comment]
Because before it was doing `hide_contacts? || !hide_contacts?`, which can only return true. I think that's what @compwron meant.

We do have a failure, however, so we'll need to check that out. Maybe just returning true was indeed needed 🤣

**Comment by compwron** [comment]
That failure may be a flake :( I re-ran it

**Comment by compwron** [comment]
Merging because only docker failed and therefore it's a flake :/

**Pull request merged**